admin_solutions/authentication.adoc

```diff
@@ -0,0 +1,166 @@
+====
++
+[NOTE]
+====
+Using `openshift start master` on its own would auto-detect hostnames, but
+GitHub must be able to redirect to the exact host name that you specified when
+registering the application. For this reason, you cannot auto-detect the ID
+because it might redirect to the wrong address. Instead, you must specify the
+hostname that web browsers use to interact with your OpenShift cluster.
+====
+. Edit the new *_master-config.yaml_* file's `*identityProviders*` stanza.
+. Copy
+link:../install_config/configuring_authentication.html#GitHub[the example `*GitHubIdentityProvider*` configuration]
+and paste it to replace the existing stanza.
+. Make the following modifications to the `*identityProviders*` stanza:
+.. Change the provider `*name*` to match the callback URL you configured on
+GitHub.
++
+For example, if you defined the callback URL as
+`https://myapiserver.com:8443/oauth2callback/github/` then the `*name*` must be
+`github`.
+.. Change the `*clientID*` to the Client ID from GitHub that you
+link:../admin_solutions/authentication.html#register-app-on-github[registered previously].
+.. Change the `*clientSecret*` to the Client Secret from GitHub that you
+link:../admin_solutions/authentication.html#register-app-on-github[registered previously].
+.. Change the `*organizations*` to include a list of one or more GitHub
+organizations to which a user must have membership in order to authenticate. If
+specified, only GitHub users that are members of at least one of the listed
+organizations will be allowed to log in. If this is not specified, then any
+person with a valid GitHub account can log in.
+. Save your changes and close the file.
+. Start the OpenShift API server, specifying the configuration file you just
+modified:
++
+----
+$ openshift start master --config=<path/to/modified/config>/master-config.yaml
+----
+
+Once configured, any user logging in to the OpenShift web console will be
+prompted to log in using their GitHub credentials. On their first login, the
+user must click *authorize application* to permit GitHub to use their username,
+password, and organization membership with OpenShift. The user is the redirected
+back to the web console.
+
+=== Creating Users
+
+A common question with people who are new to OpenShift is "How can I create
+users?"
+
+In this case, you are integrating with an external authentication provider
+(GitHub), so OpenShift is not the system of record for users. GitHub is the
+external provider of a system of record. You do not create users in OpenShift
+when you integrate with an external authenticator. Users are defined by GitHub,
+and any GitHub user belonging to a specified organization can log in.
+
+To add a user to OpenShift, you must add that user to an approved organization
+on GitHub. If required, create a new GitHub account for the user.
+
+=== Viewing Users
+
+Once one or more users have logged in, you can run `oc get users` to see a list of users:
```

Comment:
Might be good to throw in a "...to verify that users were created successfully." somewhere in here.